### ISSUE TITLE
pydantic2

### ISSUE THREAD

**Issue by joamatab** [ray-project/ray#39161]
### Description

ray[air,tune] 2.6.0 depends on pydantic<2; extra == "air"

We've been using ray as a gdsfactory plugin for chip design

however with the recent port to pydantic2 ray cannot be installed
![image](https://github.com/ray-project/ray/assets/4514346/f87ac157-44d6-411d-b382-da5a77834ecc)

See logs here https://github.com/gdsfactory/gplugins/actions/runs/6039186462/job/16387262982

**Comment by rkooo567** [comment]
cc @edoakes @alanwguo are we planning to resolve this issue by Ray 2.8?

**Comment by edoakes** [comment]
Yes aiming for 2.8. I'm picking it up this week from the Serve side

**Comment by ddelange** [comment]
ref https://github.com/ray-project/ray/pull/40451
